crates/rattler_virtual_packages/src/lib.rs

```diff
@@ -257,40 +260,95 @@ impl From<Cuda> for VirtualPackage {
-    /// the full CPU family.
-    pub spec: String,
+    /// The associated microarchitecture
+    pub spec: Arc<Microarchitecture>,
+}
+
+impl Serialize for Archspec {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        self.spec.name().serialize(serializer)
+    }
+}
+
+impl<'de> Deserialize<'de> for Archspec {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let name = String::deserialize(deserializer)?;
+        let spec = archspec::cpu::Microarchitecture::known_targets()
+            .get(&name)
+            .cloned()
+            .unwrap_or_else(|| Arc::new(archspec::cpu::Microarchitecture::generic(&name)));
+        Ok(Self { spec })
+    }
+}
+
+impl Hash for Archspec {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.spec.name().hash(state);
+    }
+}
+
+impl PartialEq<Self> for Archspec {
+    fn eq(&self, other: &Self) -> bool {
+        self.spec.name() == other.spec.name()
+    }
+}
+
+impl Eq for Archspec {}
+
+impl From<Arc<Microarchitecture>> for Archspec {
+    fn from(arch: Arc<Microarchitecture>) -> Self {
+        Self { spec: arch }
+    }
 }
 
 impl Archspec {
     /// Returns the current CPU architecture
     pub fn current() -> Option<Self> {
-        Self::from_platform(Platform::current())
+        archspec::cpu::host().ok().map(Into::into)
     }
 
-    /// Returns the CPU architecture for the given platform
+    /// Returns the minimal supported archspec architecture for the given
+    /// platform.
+    #[allow(clippy::match_same_arms)]
     pub fn from_platform(platform: Platform) -> Option<Self> {
-        let archspec = match platform {
+        // The values are taken from the archspec-json library.
+        // See: https://github.com/archspec/archspec-json/blob/master/cpu/microarchitectures.json
+        let archspec_name = match platform {
             Platform::NoArch | Platform::Unknown => return None,
-            Platform::EmscriptenWasm32 | Platform::WasiWasm32 => "wasm32",
+            Platform::EmscriptenWasm32 | Platform::WasiWasm32 => return None,
             Platform::Win32 | Platform::Linux32 => "x86",
             Platform::Win64 | Platform::Osx64 | Platform::Linux64 => "x86_64",
-            Platform::LinuxAarch64 => "aarch64",
-            Platform::LinuxArmV6l => "armv6l",
-            Platform::LinuxArmV7l => "armv7l",
+            Platform::LinuxAarch64 | Platform::LinuxArmV6l | Platform::LinuxArmV7l => "aarch64",
             Platform::LinuxPpc64le => "ppc64le",
             Platform::LinuxPpc64 => "ppc64",
             Platform::LinuxS390X => "s390x",
             Platform::LinuxRiscv32 => "riscv32",
             Platform::LinuxRiscv64 => "riscv64",
-            Platform::WinArm64 | Platform::OsxArm64 => "arm64",
+
+            // TODO: There must be a minimal aarch64 version that windows supports.
```

Comment:
Should we ask Pavel or the other person that made a PR recently in pixi about the value?

cc https://github.com/prefix-dev/pixi/pull/1053

Reply:
Im not sure if they are running the minimum requirement.

Reply:
Alright, then let's merge? :)

Reply:
I believe it ok to set it `aarch64`, Windows on ARM supports cpus down to ARMv8/armv8-a.

refs that may help:
- > The ARM64 version of Windows presupposes that it's running on an ARMv8 or later architecture at all times. Both floating-point and NEON support are presumed to be present in hardware. https://learn.microsoft.com/en-us/cpp/build/arm64-windows-abi-conventions?view=msvc-170
- > `PF_ARM_V8_INSTRUCTIONS_AVAILABLE` found in win32api https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-isprocessorfeaturepresent